apps/prairielearn/src/tests/questionSharing.test.ts

```diff
@@ -458,5 +458,22 @@ describe('Question Sharing', function () {
       );
       assert(publiclySharedQuestionRes.ok);
     });
+
+    step('Fail to sync if shared question is renamed', async () => {
+      const questionPath = path.join(sharingCourse.path, 'questions', SHARING_QUESTION_QID);
+      const questionTempPath = questionPath + '_temp';
+      await fs.rename(questionPath, questionTempPath);
+      await syncFromDisk.syncOrCreateDiskToSql(sharingCourse.path, logger);
+      const question_id = await sqldb.queryOneRowAsync(sql.get_question_id, {
+        course_id: sharingCourse.id,
+        qid: SHARING_QUESTION_QID,
+      });
+      if (!question_id) {
+        throw new Error(
+          `Sync of consuming course should not have allowed renaming a shared question.`,
+        );
+      }
```

Comment:
Two issues with this:

- `queryOneRowAsync` will always return an object, and so it's never possible for `if (!question_id)` to be true.
- `queryOneRowAsync` will throw an error if exactly one row is not found, so even if this was working how it seems like you think it is, we'll never actually perform the `if ...` check if there wasn't exactly one row.

My recommendation: use `queryOptionalRow` instead and pair it with `assert(..., 'Sync of consuming course should not allow renaming a shared question.')`.

Reply:
that's emberassing :sweat_smile: . I'll fix it